openlibrary/plugins/openlibrary/js/validate.js

```diff
@@ -20,11 +20,15 @@ export default function initValidate() {
     // validate publish-date to make sure the date is not in future
     // used in templates/books/add.html
     jQuery.validator.addMethod('publish-date', function(value) {
+
+        // we allow the year dates 'xxxx', '199x', '19xx' date variants: https://github.com/internetarchive/openlibrary/issues/5254
+        var token_xxxx = /(\b[1-9|x][1-9|x][1-9|x][1-9|x]\b)/.exec(value);
+
         // if it doesn't have even three digits then it can't be a future date
         var tokens = /(\d{3,})/.exec(value);
```

Comment:
```suggestion
        var tokens = /^(\d+[x]{0,2}|[x]{4})$/.exec(value);
```
We could try something like this, but then there is the issue of future dates when it comes to the return value. E.g., the following returns `true`, because `parseInt("30xx")` returns `30`:
```
var tokens = /^(\d+[x]{0,2}|[x]{4})$/.exec("30xx")
tokens && tokens[1] && (parseInt(tokens[1]) <= year + 1 || tokens[1].startsWith("x"));
```

Validating dates with `x` starts to make this a bit more complex (unless Mr. Algorithm, @cdrini has some ideas). Perhaps we could simply replace any `x` characters with `9` when checking if something is published after the `currentYear + 1`. Granted, that would mean `20xx` and `202x` fail the future-publication check, but perhaps it makes sense to disallow approximate years in that are so recent.

Thoughts?

Reply:
We could also have different tokens for the 'xxxx' case vs the '20xx' and '199x' type cases too.